src/test/mod.rs

```diff
@@ -84,7 +84,21 @@ impl<'a> ModuleRenderer<'a> {
 
     /// Renders the module returning its output
     pub fn collect(self) -> Option<String> {
-        crate::print::get_module(self.name, self.context)
+        let ret = crate::print::get_module(self.name, self.context);
+        // all tests rely on the fact that an empty module produces None as output as the
+        // convention was that there would be no module but None. This is nowadays not anymore
+        // the case (to get durations for all modules). So here we make it so, that an empty
+        // module returns None in the tests...
+        match &ret {
```

Comment:
I'd prefer something like this:
```rust
match ret {
    Some("") => None,
    Some(_) => ret,
    None => None,
}
```

Reply:
better yet:
```rust
match ret {
    Some("") => None,
    _ => ret,
}
```
or
```rust
match ret {
    Some("") => None,
    ret => ret,
}
```

Reply:
Tried to do it but rust doesn't like it: `ret` is a `Option<String>` but `Some("")` is a `Option<str>`. And `Some(String::from(""))` is an illegal function call in the match and matching `ret.ref()` doesn't work because ret is an Option  :-(

The best I could come up with is 

``` 
if ret.is_none() {
            return None
        } else {
            let s = ret.unwrap();
            match s.as_str() {
                "" => None,
                _ => Some(s),
            }
        }
``` 

and this doesn't look much better than what is in there now :-(

Reply:
Sorry I just ran `cargo check` and I forgot that doesnt include this function, try: `ret.and_then(|s| if s == "" { None } else { Some(s) })`

Reply:
Wait there's also `filter`: `ret.filter(|s| s != "")`

Reply:
Wow, nice!

Reply:
done